Sources/Hexiconjuror/Commands/Output Strings/OutputStrings.swift

```diff
@@ -0,0 +1,221 @@
+                try localize.run()
+                localize.waitUntilExit()
+            }
+            .flatMapCatching {
+                let existingTables = try stringsFiles.map {
+                    try Table(url: $0, strings: parser.parse(file: $0))
+                }
+                let newTables = try FileManager.default.contentsOfDirectory(at: temporaryDirectory, includingPropertiesForKeys: [.isRegularFileKey], options: [])
+                    .filter { $0.pathExtension == "strings" }
+                    .map { return try Table(url: $0, strings: parser.parse(file: $0)) }
+                return Array(newTables.map { newTable -> [Table] in
+                    let matches = existingTables.filter { $0.name == newTable.name }
+                    guard !matches.isEmpty else {
+                        return [mutate(newTable) { $0.url = resourcesPath.appendingPathComponent("\($0.name).strings") }]
+                    }
+                    return matches.map {
+                        var matchingTable = $0
+                        newTable.strings.forEach { newString in
+                            guard var string = matchingTable.strings[newString.key] else {
+                                matchingTable.strings[newString.key] = newString.value
+                                return
+                            }
+                            if string.hasEmptyComment {
+                                string.comment = newString.value.comment
+                            }
+                            matchingTable.strings[newString.key] = string
+                        }
+                        matchingTable.strings = matchingTable.strings.filter {
+                            newTable.strings[$0.key] != nil
+                        }
+                        return matchingTable
+                    }
+                }
+                    .joined())
+            }
+            .flatMapCatching { (tables: [Table]) in
+                try tables.forEach { table in
+                    let oldFile = temporaryDirectory.appendingPathComponent(UUID().uuidString)
+                    let tempFile = temporaryDirectory.appendingPathComponent(UUID().uuidString)
+                    FileManager.default.createFile(atPath: tempFile.path, contents: nil, attributes: nil)
+                    try table.write(to: tempFile, stripEmptyComments: options.stripEmptyComments)
+                    if FileManager.default.fileExists(atPath: table.url.path) {
+                        try FileManager.default.moveItem(at: table.url, to: oldFile)
+                    }
+                    do {
+                        try FileManager.default.moveItem(at: tempFile, to: table.url)
+                    } catch {
+                        try FileManager.default.moveItem(at: oldFile, to: table.url)
+                    }
+                }
+            }
+            .mapError { _ in ConjurorError.invalidArguments }
```

Comment:
I think it would be desirable to give some information about the inner error as well, e.g.: an error thrown on line 42 or 98. True they probably occur because the arguments are invalid, but the system error might have more information about how to fix it?

Reply:
I'm going to open an issue on adding actual errors. IIRC the way Commandant is set up makes it kind of painful where every error needs to be the same type so I avoided doing anything drastic, but I think with enough effort one could make a generic wrapper error and each command could have its own error which coalesces into the generic wrapper automatically, satisfying logical soundness and type constraints.